src/foam/mlang/MLang.java

```diff
@@ -82,7 +82,7 @@ public static Sink MAX(Object o1) {
   }
 
   public static Sink MIN(Object o1) {
-    return new Min(0, MLang.prepare(o1));
+    return new Min(Double.MAX_VALUE, null, MLang.prepare(o1));
```

Comment:
I think it would be better to make the initial value be null and then have the compare take the first put() value as the min value if the current value is null.